plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TrinoHiveCatalog.java

```diff
@@ -215,7 +219,26 @@ public boolean dropNamespace(ConnectorSession session, String namespace)
                 !listViews(session, Optional.of(namespace)).isEmpty()) {
             throw new TrinoException(SCHEMA_NOT_EMPTY, "Schema not empty: " + namespace);
         }
-        metastore.dropDatabase(new HiveIdentity(session), namespace, true);
+
+        Optional<Path> location = metastore.getDatabase(namespace)
+                .orElseThrow(() -> new SchemaNotFoundException(namespace))
+                .getLocation()
+                .map(Path::new);
+
+        // If we see files in the schema location, don't delete it.
+        // If we see no files or can't see the location at all, use fallback.
```

Comment:
> If we see no files

then we delete the directory

> or can't see the location at all, use fallback.

only on this latter case we use fallback

did you mean "when location is not set"?

Reply:
If we see no files, delete. If we can't see or if `getDatabase` didn't get the location (it always should in practice, even when using default location) use fallback. I'll update the comment.